### PR TITLE
Migrate `src/runtime/pay-flow` to TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**

### CODE REVIEW

File: src/runtime/pay-flow-test.js [hunk 1]
```diff
@@ -1054,6 +1054,80 @@ describes.realWin('PayCompleteFlow', (env) => {
     await flow.readyPromise_;
   });
 
+  it('constructs valid flow w/ virtual gift url params even without order id', async () => {
```
Comment (fikus) [inline src/runtime/pay-flow-test.js:1057]: virtual gifts is TWG, so there's probably no need for a new test.

File: src/runtime/pay-flow.ts [hunk 2]
```diff
@@ -586,59 +541,49 @@ export function parseSubscriptionResponse(deps, data, completeHandler) {
...
 /**
- * @param {!Object} swgData
- * @return {?UserData}
- * @package Visible for testing.
+ * Visible for testing.
  */
-export function parseUserData(swgData) {
-  const idToken = swgData['idToken'];
+export function parseUserData(swgData: object): UserData | null {
+  const idToken = (swgData as {idToken?: string})['idToken'];
```
Comment (fikus) [inline src/runtime/pay-flow.ts:554]: Can you type `swgData` in the function signature instead, or does that break any callers?

File: src/runtime/pay-flow.ts [hunk 3]
```diff
@@ -586,59 +541,49 @@ export function parseSubscriptionResponse(deps, data, completeHandler) {
...
- * @package Visible for testing.
+ * Visible for testing.
  */
-export function parseEntitlements(deps, swgData) {
-  if (swgData['signedEntitlements']) {
+export function parseEntitlements(
+  deps: Deps,
+  swgData: object
+): Entitlements | null {
+  if ((swgData as {signedEntitlements?: string})['signedEntitlements']) {
```
Comment (fikus) [inline src/runtime/pay-flow.ts:569]: Same question here about swgData.

**Review by fikus — APPROVED** [review]

File: src/runtime/pay-flow-test.js (hunk 1 above)
Comment (ChrisAntaki) [reply src/runtime/pay-flow-test.js:1057]: The TypeScript migration is exposing branches that were missing coverage. If code relating to `VIRTUAL_GIFT` purchases is still necessary, I think testing all the branches makes sense. If it's dead code though, we can remove it in a followup PR.

File: src/runtime/pay-flow.ts (hunk 2 above)
Comment (ChrisAntaki) [reply src/runtime/pay-flow.ts]: Good call, updated

File: src/runtime/pay-flow.ts (hunk 3 above)
Comment (ChrisAntaki) [reply src/runtime/pay-flow.ts]: Updated

**Pull request merged**